### PR TITLE
Backend of interests feature

### LINKED ISSUE #1781: Add Interests Feature (backend)
Create model and API views to make users select causes of interests.

**Actions needed**
- Create database model.
- Create services to create, delete and update interests.
- Enable users to select multiple types of interests.
- associate projects with the Interests model.
- Compute user contributions rate grouped by project interests

### PR DESCRIPTION
**Pull request by JorgeMartinezG**
This PR closes #1781, which is the cause of interests feature.

**How to test**
1. Create a cause of interest: `POST /api/v1/interests` with the following payload
{
  "name": "string"
}

2. Get all interests: GET `/api/v1/interests`
3. Create a relationship between project and interests: POST `/api/v1/interests/project` with the following payload
{
  "interests": [1, 2, 3, ..., n],
  "project_id": int
}
4. Creates a relationship between user and interests: POST `/api/v1/interests/user`
{
  "interests": [1, 2, 3, ..., n],
  "user_id": int
}
5. Get rate of contributions from a user given his/her interests: GET `/api/v1/interests/user/{user_id}`
6. Update interest name: PUT `/api/v1/interests/{interest_id}` with the following payload:
{
  "name": "string"
}

**Note.** I also created a management command to load multiple causes of interests from a file. Included within test suite also. ❤️

### CODE REVIEW

**Comment by JorgeMartinezG** [comment]
Endpoints have changed.

1. Create a cause of interest: POST /api/v1/interests with the following payload
{
"name": "string"
}

2. Get all interests: GET /api/v1/interests

Create a relationship between project and interests: POST /api/v2/projects/actions/set-interests with the following payload
{
"interests": [1, 2, 3, ..., n],
"project_id": int
}

Creates a relationship between user and interests: POST /api/v2/projects/users/set-interests
{
"interests": [1, 2, 3, ..., n],
"user_id": int
}

Get rate of contributions from a user given his/her interests: GET /api/v2/users/queries/interests/<int:user_id>

Update interest name: PATCH /api/v1/interests/{interest_id} with the following payload:
{
"name": "string"
}

**Comment by JorgeMartinezG** [comment]
@xamanu Implemented requested changes.

**Comment by xamanu** [comment]
@JorgeMartinezG Can you please resolve the conflicts. Thank you.

File: manage.py [hunk 1]
```diff
@@ -58,5 +59,10 @@ def refresh_levels():
     print(f"Updated {users_updated} user mapper levels")
 
 
+@manager.option("-f", "--file", help="JSON file")
```
[the remark this answers is not in the document]
Comment (xamanu) [reply manage.py]: I created an issue for the UI: #1926

**Comment by xamanu** [comment]
There are still some open comments. Please answer or code on these :smile: thanks.

**Comment by xamanu** [comment]
Giving back to @JorgeMartinezG for rebasing on latest `develop` and fixing some conflicts. Thanks.

**Comment by JorgeMartinezG** [comment]
@willemarcel thanks for catching this bug. Verified and fix sent

**Comment by willemarcel** [comment]
Thanks, @JorgeMartinezG !

**Pull request merged**